### PR TITLE
Fusion: Install Pyside for Fusion16+

### PR DESCRIPTION
**Pull request by EmberLightVFX**
Install pyside now works for all Fusion Version.
It also installs pyside6 instead of pyside2

This is a re-pr of an OpenPype pr: https://github.com/ynput/OpenPype/pull/4932

## Testing notes:
1. Start fusion with a python instance that doesn't have Pyside installed
2. Run `Install pyside` from the Ayon menu

### CODE REVIEW

File: client/ayon_core/hosts/fusion/deploy/MenuScripts/install_pyside.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# This is just a quick hack for users running Py3 locally but having no
+# Qt library installed
+import os
+import importlib
+from pathlib import Path
+import platform
+import subprocess
+from openpype.hosts.fusion import FUSION_VERSIONS_DICT
```
Comment (jakubjezek001) [inline client/ayon_core/hosts/fusion/deploy/MenuScripts/install_pyside.py:8]:
```suggestion
from ayon_core.hosts.fusion import FUSION_VERSIONS_DICT
```

**Review by jakubjezek001 — CHANGES_REQUESTED** [review]
Thank you @EmberLightVFX for the code changes. Here are my suggestions for improvement.

*   Remove the file `\fusion\hooks\pre_pyside_install.py`. It automatically installs PySide2 if it's not found, preventing issues with missing PySide.
*   Add a pop-up message to tell users they need to install PySide through the menu if it's not on their computer. Right now, it just crashes with a confusing error message.
*   Fix the bug that occurs when a user fixes the missing PySide6 with the install script and then tries to start the AYON menu in the same Fusion session. The following traceback happens. This was not happening for the remaining PySide2 reason from first point.

```Python
Traceback (most recent call last):
  File "C:\CODE\__PYPE\ayon-core\client\ayon_core\hosts\fusion\deploy\MenuScripts\launch_menu.py", line 47, in <module>
    result = main(os.environ)
  File "C:\CODE\__PYPE\ayon-core\client\ayon_core\hosts\fusion\deploy\MenuScripts\launch_menu.py", line 38, in main
    menu.launch_ayon_menu()
  File "C:\CODE\__PYPE\ayon-core\client\ayon_core\hosts\fusion\api\menu.py", line 180, in launch_ayon_menu
    ayon_menu = AYONMenu()
  File "C:\CODE\__PYPE\ayon-core\client\ayon_core\hosts\fusion\api\menu.py", line 77, in __init__
    layout.addWidget(context_label)
TypeError: 'PySide6.QtWidgets.QBoxLayout.addWidget' called with wrong argument types:
  PySide6.QtWidgets.QBoxLayout.addWidget(QLabel)
Supported signatures:
  PySide6.QtWidgets.QBoxLayout.addWidget(PySide6.QtWidgets.QWidget, int = 0, PySide6.QtCore.Qt.AlignmentFlag = Default(Qt.Alignment))
  ```

**Comment by jakubjezek001** [comment]
This comment is not related to this PR but I have just noticed something during testing of this PR.

Perhaps it is happening also to you all @BigRoy and @EmberLightVFX. For some reason any log.info messages are showing up in Fusion console as warnings and it is causing a confusion because termminal icon is dressed with exclamation mark :warning:

File: client/ayon_core/hosts/fusion/deploy/MenuScripts/install_pyside.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+except Exception:
+    # app is a predefined variable for menu scripts
+    app = app  # noqa: F821
+    fusion_version = int(app.Version)
+    fusion_python_home, _ = FUSION_VERSIONS_DICT.get(fusion_version)
+    fusion_python_home = os.environ.get(fusion_python_home)
+    print("Assuming no Qt library is installed..")
+    print("Installing PySide package for " f"{fusion_python_home}")
+    pip_install("PySide6", fusion_python_home)
```
Comment (iLLiCiTiT) [inline client/ayon_core/hosts/fusion/deploy/MenuScripts/install_pyside.py:79]:
```suggestion
    pip_install("PySide6==6.6.3.1", fusion_python_home)
```

Comment (jakubjezek001) [reply client/ayon_core/hosts/fusion/deploy/MenuScripts/install_pyside.py:79]: but should this be compatible with every Fusion version?

Comment (iLLiCiTiT) [reply client/ayon_core/hosts/fusion/deploy/MenuScripts/install_pyside.py:79]: True PySide6 is available since Python 3.9

**Comment by iLLiCiTiT** [comment]
Fusion has been moved to separate [https://github.com/ynput/ayon-fusion](repository). Please create PR there if is still relevant.

**Pull request closed without merging**